### PR TITLE
Fixed Conflict

### PR DESCRIPTION
**Pull request by Tadas-M**

### CODE REVIEW

File: University-advisor/ApiCalls.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+namespace University_advisor
+{
+    class ApiCalls {
+
+        public string GetLocationJson(string url) {
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+            try
+            {
+                WebResponse response = request.GetResponse();
+                using (Stream responseStream = response.GetResponseStream())
```
Comment (avizen-j) [inline University-advisor/ApiCalls.cs:19]: I guess it is better to use implicitly typed variables, so just declare it as var responseSteam.

File: University-advisor/ApiCalls.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace University_advisor
+{
+    class ApiCalls {
+
+        public string GetLocationJson(string url) {
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+            try
+            {
+                WebResponse response = request.GetResponse();
```
Comment (avizen-j) [inline University-advisor/ApiCalls.cs:18]: Just use var response.

File: University-advisor/ApiCalls.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+using System.Net;
+using System.IO;
+using System.Web.Script.Serialization;
+
+namespace University_advisor
+{
+    class ApiCalls {
+
+        public string GetLocationJson(string url) {
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
```
Comment (avizen-j) [inline University-advisor/ApiCalls.cs]: Use implicitly typed variables. Same goes for line 22, 28, 29, 31, 32

File: University-advisor/Geocoding.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+using System.Device.Location;
+using System.Globalization;
+using University_advisor.Models;
+using Newtonsoft.Json;
+
+namespace University_advisor
+{
+    class Geocoding
+    {
+        string apiKey = "5e66cc9d64db23";
```
Comment (avizen-j) [inline University-advisor/Geocoding.cs]: In this class you were declaring your variables as var, why did you make an exception for this line?

File: University-advisor/Geocoding.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            var locationDataA = JsonConvert.DeserializeObject<List<LocationData>>(jsonA);
+            var locationDataB = JsonConvert.DeserializeObject<List<LocationData>>(jsonB);
+
+            //Using first member of array because there can be multiple returned locations
+            var locationA = new Geocoding().GetCoordinates(locationDataA[0]);
+            var locationB = new Geocoding().GetCoordinates(locationDataB[0]);
+
+            var distance = new Geocoding().GetDistance(locationA.Item1, locationA.Item2, locationB.Item1, locationB.Item2);
+            //Console.WriteLine(distance);
+            //Console.ReadLine();
```
Comment (avizen-j) [inline University-advisor/Geocoding.cs]: Delete unnecessary comments.

File: University-advisor/Geocoding.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+            return distance;
+        }
+
+        public (double, double) GetCoordinates(LocationData address)
```
Comment (avizen-j) [inline University-advisor/Geocoding.cs:32]: Not a mistake but I am wondering how this actually works? Method returns two double results, how do you select which one to choose when calling it?

**Review by avizen-j — CHANGES_REQUESTED** [review]
Code is clean and well - readable however some minor refactoring has to be done.

File: University-advisor/ApiCalls.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            catch(WebException ex)
+            {
+                WebResponse errorResponse = ex.Response;
+                using (Stream responseStream = errorResponse.GetResponseStream())
+                {
+                    StreamReader reader = new StreamReader(responseStream, Encoding.GetEncoding("utf-8"));
+                    string errorText = reader.ReadToEnd();
+                    //TO DO: Log error
+                }
+                throw;
```
Comment (avizen-j) [inline University-advisor/ApiCalls.cs:34]: What is the point of throwing exception in catch block?

File: University-advisor/Geocoding.cs (hunk 6 above)
Comment (Tadas-M) [reply University-advisor/Geocoding.cs:32]: With this expression you receive tuple as returned object

>  var locationA = new Geocoding().GetCoordinates(locationDataA[0]);
    var locationB = new Geocoding().GetCoordinates(locationDataB[0]);
 var distance = new Geocoding().GetDistance(locationA.Item1, locationA.Item2, locationB.Item1, locationB.Item2);

As you can see here, the returned object has properties item1 and item2 which are those 2 values

Comment (avizen-j) [reply University-advisor/Geocoding.cs:32]: Great feature, thank you for explaining it.

**Comment by avizen-j** [comment]
Geocoding api integration.

**Pull request merged**